lib/ansible/modules/extended_mount_facts.py

```diff
@@ -0,0 +1,282 @@
+"""
+
+RETURN = """
+ansible_facts:
+    description: An ansible_facts dictionary containing C(extended_mounts).
+    returned: always
+    type: dict
+    sample:
+      extended_mounts:
+        /mnt/mount:
+          info:
+            block_available: 3242510
+            block_size: 4096
+            block_total: 3789825
+            block_used: 547315
+            device: hostname:/srv/sshfs
+            fstype: fuse.sshfs
+            inode_available: 1875503
+            inode_total: 1966080
+            mount: /mnt/mount
+            options: "rw,nosuid,nodev,relatime,user_id=0,group_id=0"
+            size_available: 13281320960
+            size_total: 15523123200
+          uuid: N/A
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.facts import timeout
+from ansible.module_utils.facts.hardware.linux import LinuxHardware
+from ansible.module_utils.facts.utils import get_mount_size, get_file_content
+
+from typing import Callable, Optional
+
+import re
+
+# Regular expressions to match individual lines from mount
+
+# FreeBSD: hostname:/srv/sshfs on /mnt/sshfs (fusefs.sshfs)
+# MacOSX: //user@hostname/public on /mnt/cifs (smbfs, nodev, nosuid, mounted by user)
+BSD_MOUNT_RE = re.compile(r"^(?P<device>\S+) on (?P<mount>\S+) \((?P<fstype>.+)\)$")
+
+# AIX: hostname /mount /mounted_over nfs Dec 17 08:06 ro, log  =/dev/hd8
+AIX_MOUNT_RE = re.compile(r"^(?P<node>\S+)\s+(?P<mount>\S+)\s+(?P<mounted_over>\S+)\s+(?P<fstype>\S+)\s+(?P<time>\S+\s+\d+\s+\d+:\d+)\s+(?P<options>.*)$")
+
+# TODO...?
+
+# No mount parsing preferably
+# Fedora, Debian, Ubuntu, Arch use /etc/mtab or /proc/mounts
+# Solaris uses /etc/mnttab
+
+
+TYPES = [
+    "cifs",
+    "ceph",
+    "fuse",
+    "glusterfs",
+    "gpfs",
+    "nfs",
+    "nfs4",
+    "vboxsf",
+    "zfs",
+]
+
+
+def get_mount_info(
+    module: AnsibleModule,
+    mount: str,
+    device: str,
+    uuids: dict[str, str],
+    udevadm_uuid: Callable[[str], str]
+) -> tuple[Optional[dict], Optional[str]]:
+    """
+    Attempts to get the mount size and UUID within the specified timeout.
+    """
+    seconds = module.params["timeout"] or .01
+    on_error = module.params["on_timeout"]
+
+    @timeout.timeout(seconds)
+    def _get_mount_info(mount, device, uuids, udevadm_uuid):
+        mount_size = get_mount_size(mount)
+        uuid = uuids.get(device, udevadm_uuid(device))
+        return mount_size, uuid
+
+    try:
+        return _get_mount_info(mount, device, uuids, udevadm_uuid)
+    except timeout.TimeoutError:
+        if on_error == "error":
+            module.fail_json(msg=f"{device} on {mount} timed out")
+        elif on_error == "warn":
+            module.warn(f"{device} on {mount} timed out")
+    return None, None
+
+
+def parse_mount(line: str) -> Optional[dict[str, str]]:
+    """
+    Parse a line of mount information and return a dictionary with mount details.
+    """
+    match = BSD_MOUNT_RE.match(line) or AIX_MOUNT_RE.match(line)
```

Comment:
should we check OS/platform first in case some weird fs driver can match 'other platforms'?

Reply:
I added a check for sys.platform, but made it kind of lenient (allowing 'linux' to match) since it seems there is no standardization.

Reply:
sorry, it was a bad idea on my part, it would need to include the distro lib (we do vendor) and a lot of complicated and fallback heavy querying. I forgot that 'platform detection' is a mess.